tower-sessions-core/src/session.rs

```diff
@@ -552,13 +564,13 @@ impl Session {
     /// ```
     pub fn expiry_date(&self) -> OffsetDateTime {
         let expiry = self.expiry.lock();
+        let now = now_utc();
```

Comment:
This is a good opportunity to break this apart into its own function to better facilitate testing.

For example, something along these lines:

```diff
diff --git a/tower-sessions-core/src/session.rs b/tower-sessions-core/src/session.rs
index ad0cd93..dfa49d2 100644
--- a/tower-sessions-core/src/session.rs
+++ b/tower-sessions-core/src/session.rs
@@ -551,14 +551,18 @@ impl Session {
     /// assert!(session.expiry_date() < expected_expiry.saturating_add(Duration::seconds(1)));
     /// ```
     pub fn expiry_date(&self) -> OffsetDateTime {
+        self.expiry_date_with_now(OffsetDateTime::now_utc())
+    }
+
+    fn expiry_date_with_now(&self, now: OffsetDateTime) -> OffsetDateTime {
         let expiry = self.expiry.lock();
         match *expiry {
-            Some(Expiry::OnInactivity(duration)) => {
-                OffsetDateTime::now_utc().saturating_add(duration)
-            }
+            Some(Expiry::OnInactivity(duration)) => now.saturating_add(duration),
             Some(Expiry::AtDateTime(datetime)) => datetime,
             Some(Expiry::OnSessionEnd) | None => {
-                OffsetDateTime::now_utc().saturating_add(DEFAULT_DURATION) // TODO: The default should probably be configurable.
+                now.saturating_add(DEFAULT_DURATION) // TODO: The default should
+                                                     // probably be
+                                                     // configurable.
             }
         }
     }
```

Reply:
Seems like you are suggesting a dep injection style. Again, I have no problem with that. I was just trying to do the minimum.

Again, this is fine.